lib/api/src/rc_get_resource_info_for_operation.cpp

```diff
@@ -0,0 +1,18 @@
+#include "irods/get_resource_info_for_operation.h"
+
+#include "irods/procApiRequest.h"
+#include "irods/apiNumber.h"
+
+auto rc_get_resource_info_for_operation(RcComm* _comm, const DataObjInp* _dataObjInp, char** _out_info) -> int
+{
+        if (!_comm || !_dataObjInp || !_out_info) {
+                return SYS_INVALID_INPUT_PARAM;
+        }
```

Comment:
Let's add a check like this on the server side as well for full belt and suspenders